### PR TITLE
REF: use _validate_foo pattern in Categorical

### PR DESCRIPTION
**Pull request by jbrockmendel**
Similar to how we did it in datetimelike, eventually these validator methods are going to end up sharing most of their code

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -1192,6 +1192,26 @@ def map(self, mapper):
...
+    def _validate_insert_value(self, value) -> int:
+        code = self.categories.get_indexer([value])
+        if (code == -1) and not (is_scalar(value) and isna(value)):
+            raise TypeError(
+                "cannot insert an item into a CategoricalIndex "
+                "that is not already an existing category"
+            )
+        return code[0]
+
+    def _validate_searchsorted_value(self, value):
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:1204]: can you type as much as possible

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1204]: sure, there are a few of these and im planning to do a typing pass at the end

Comment (jreback) [reply pandas/core/arrays/categorical.py:1204]: kk

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
The related others are  #36173 (just rebased), #36180, #36171, #36176

**Pull request merged**